docs/source/generating/testing.rst

```diff
@@ -233,6 +233,34 @@ Or we can simplify further by using ``assertHasMessage`` and ``assertHasAction``
             self.assertEqual(servers, [msg.message["server"] for msg in messages])
 
 
+Custom JSON encoding
+--------------------
+
+Just like a ``FileDestination`` can have a custom JSON encoder, so can your tests, so you can validate your messages with that JSON encoder:
+
+.. code-block:: python
+
+   from unittest import TestCase
+   from eliot.json import EliotJSONEncoder
+   from eliot.testing import capture_logging
+
+   class MyClass:
+       def __init__(self, x):
+           self.x = x
+
+   class MyEncoder(EliotJSONEncoder):
+       def default(self, obj):
+           if isinstance(obj, MyClass):
+               return {"x": obj.x}
+           return EliotJSONEncoder.default(self, obj)
+
+   class LoggingTests(TestCase):
+       @capture_logging(None, _encoder=MyEncoder)
```

Comment:
I assume it is named `_encoder` because `capture_logging` takes `**kwargs` and `encoder` may be more likely than `_encoder` to collide with an application-defined keyword argument.  If so, yea, I don't really see a better option - apart from introducing a replacement for `capture_logging` that doesn't have the problem.

It might be worth being explicit about this rationale and making it clear that `_encoder` is meant to be a part of the public Eliot testing API.

Alternatively, it could be `encoder_` - presumably at least as unlikely to collide but without the privacy implications?

Reply:
Yeah, not really excited about a new API this month, at least. `encoder_` seems nicer, will switch.